File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        unordered_set<char> vowels_set = {'a', 'e', 'i', 'o','u', 'A', 'E', 'I', 'O', 'U'};

        int n = s.size();

        int l = 0;
        int r = n - 1;

        while (l < r)
        {
           while (l < r && !vowels_set.count(s[l]))
             l++;

           while (l < r && !vowels_set.count(s[r]))
             r--;

           if (l < r)
           {
              swap(s[l], s[r]);
              l++;
              r--;
           }

        }

       return s;
    }
};
 

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        int n = s.length();
        int l = 0;
        int r = n - 1;
       
        char[] arr = s.toCharArray();
        Set<Character> vowels_set = new HashSet<>(Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));

        while (l < r)
        {
           while (l < r && !vowels_set.contains(s.charAt(l)))
             l++;

           while (l < r && !vowels_set.contains(s.charAt(r)))
             r--;

            if (l < r)
            {
               char t = arr[l];
               arr[l] = arr[r];
               arr[r] = t;

               l++;
               r--;
            }
        }
        return new String(arr);
    }
}